bump-version.sh

```diff
@@ -16,17 +17,30 @@ case "$1" in
     ;;
 
   *)
-    v2=$1
+
+    # Trim off preceding `v` if any.
+    # This is to support both input version formats: `0.1.0` and `v0.1.0`.
+    v2=$(echo "$1" | sed -Ee 's/^v//')
     ;;
 esac
 
 echo "$v => $v2"
 
 for f in $VERSIONED_FILES; do \
+  prevCount=$(grep -c -i "$v2" "$f")
+
+  # Replace the version.
   echo "- $f"; \
   if [[ "$OSTYPE" == "darwin"* ]]; then
     sed -i '' "s/$v/$v2/g" "$f"; \
   else
     sed -i "s/$v/$v2/g" "$f"; \
   fi
+
+  # Check if the version has being properly replaced.
+  newCount=$(grep -c -i "$v2" "$f")
+  if [[ ! $newCount > $prevCount ]]; then
```

Comment:
`>` is for string comparisons, use `-gt` instead https://www.shellcheck.net/wiki/SC2071

Reply:
good point! updated in 51ba8e204a85b4a3d4f0d6d29a68fcc458f10ec7